lib/player.js

```diff
@@ -1356,6 +1356,13 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         this.playhead_ = null;
       }
 
+      // EME v0.1b requires the media element to clear the MediaKeys
+      if (shaka.util.Platform.isMediaKeysPolyfilled('webkit') &&
+          this.drmEngine_) {
+        await this.drmEngine_.destroy();
```

Comment:
Is there a downside to destroying DRM engine this early in all cases?  Do we need two paths for this?

Reply:
We wanted to make this patch not invasive as much as possible. But I agree though, having diverged paths is not elegant. I can try to move drm engine teardown before the MSE one.

Reply:
@tykus160 please create a feat with the change

Reply:
Hey @joeyparrish @avelad 
According to EME spec:
> Support for clearing or replacing the associated [`MediaKeys`](https://w3c.github.io/encrypted-media/#dom-mediakeys) object during playback is a quality of implementation issue. In many cases it will result in a bad user experience or rejected promise.
In Chrome if I move drm destroy before mse, next playbacks of protected content are starting to stall after few secs. I’m using Angel One to test, I guess first few secs are clear there... I'm not sure why it is like that, but I see that `MediaKeys` are attached to the video element.
Explicitly unloading video element in DRM Engine before setting `MediaKeys` to `null` seems to fix the issue mentioned above and I see no drawbacks across Chrome, Firefox & Safari in MSE mode. I need to check though how it behaves on SmartTVs.